ai_flow/scheduler/worker.py

```diff
@@ -14,24 +14,136 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+import json
+import logging
 import queue
 import threading
 
+from notification_service.event import Event
+from notification_service.notification_client import NotificationClient
+
+from ai_flow.common.util.db_util.session import create_session
+from ai_flow.metadata.metadata_manager import MetadataManager
+from ai_flow.model.action import TaskAction
+from ai_flow.model.internal.events import SchedulingEventType, EventContextConstant, HAS_WORKFLOW_EXECUTION_ID_SET
+from ai_flow.model.status import TaskStatus, TASK_ALIVE_SET
+from ai_flow.scheduler.rule_executor import RuleExecutor
+from ai_flow.scheduler.rule_wrapper import WorkflowRuleWrapper
+from ai_flow.scheduler.schedule_command import WorkflowExecutionStartCommand, WorkflowExecutionStopCommand, \
+    WorkflowExecutionScheduleCommand
+from ai_flow.scheduler.scheduling_event_processor import SchedulingEventProcessor
 from ai_flow.scheduler.scheduling_unit import SchedulingUnit
+from ai_flow.scheduler.workflow_executor import WorkflowExecutor
 
 
 class Worker(threading.Thread):
     """Worker's responsibility is executing the scheduling unit."""
 
     def __init__(self,
                  max_queue_size: int = 20,
-                 task_executor=None):
+                 task_executor=None,
+                 db_engine=None,
+                 notification_client: NotificationClient = None):
         super().__init__()
         self.input_queue = queue.Queue(max_queue_size)
         self.task_executor = task_executor
+        self.db_engine = db_engine
+        self.notification_client = notification_client
 
     def add_unit(self, unit: SchedulingUnit):
         self.input_queue.put(unit)
 
+    def _execute_workflow_execution_schedule_command(self,
+                                                     command: WorkflowExecutionScheduleCommand,
+                                                     metadata_manager: MetadataManager):
+        for c in command.task_schedule_commands:
+            self.task_executor.schedule_command(c)
+            # set the task execution status
+            if TaskAction.START == c.action or TaskAction.RESTART == c.action:
+                task_execution_meta = metadata_manager.get_task_execution(
+                    workflow_execution_id=c.new_task_execution.workflow_execution_id,
+                    task_name=c.new_task_execution.task_name,
+                    sequence_number=c.new_task_execution.seq_num)
```

Comment:
I remember when TaskAction == START, `new_task_execution` is None while `current_task_execution` carries meta.